server/src/test/java/org/opensearch/cluster/routing/remote/RemoteRoutingTableServiceTests.java

```diff
@@ -0,0 +1,491 @@
+        assertEquals(1, diff.getUpserts().size());
+        assertNotNull(diff.getUpserts().get(indexName));
+        assertEquals(noOfShards + 1, diff.getUpserts().get(indexName).getShards().size());
+        assertEquals(noOfReplicas + 2, diff.getUpserts().get(indexName).getShards().get(0).getSize());
+
+        assertEquals(0, diff.getDeletes().size());
+    }
+
+    public void testGetIndicesRoutingMapDiffIndexDeleted() {
+        String indexName = randomAlphaOfLength(randomIntBetween(1, 50));
+        final IndexMetadata indexMetadata = new IndexMetadata.Builder(indexName).settings(
+            Settings.builder()
+                .put(IndexMetadata.SETTING_VERSION_CREATED, Version.CURRENT)
+                .put(IndexMetadata.SETTING_INDEX_UUID, "uuid")
+                .build()
+        ).numberOfShards(randomInt(1000)).numberOfReplicas(randomInt(10)).build();
+        RoutingTable routingTable = RoutingTable.builder().addAsNew(indexMetadata).build();
+
+        String indexName2 = randomAlphaOfLength(randomIntBetween(1, 50));
+        final IndexMetadata indexMetadata2 = new IndexMetadata.Builder(indexName2).settings(
+            Settings.builder()
+                .put(IndexMetadata.SETTING_VERSION_CREATED, Version.CURRENT)
+                .put(IndexMetadata.SETTING_INDEX_UUID, "uuid2")
+                .build()
+        ).numberOfShards(randomInt(1000)).numberOfReplicas(randomInt(10)).build();
+        RoutingTable routingTable2 = RoutingTable.builder().addAsNew(indexMetadata2).build();
+
+        DiffableUtils.MapDiff<String, IndexRoutingTable, Map<String, IndexRoutingTable>> diff = RemoteRoutingTableService
+            .getIndicesRoutingMapDiff(routingTable, routingTable2);
+        assertEquals(1, diff.getUpserts().size());
+        assertNotNull(diff.getUpserts().get(indexName2));
+
+        assertEquals(1, diff.getDeletes().size());
+        assertEquals(indexName, diff.getDeletes().get(0));
+    }
+
+    public void testGetIndexRoutingAsyncAction() throws IOException {
+        String indexName = randomAlphaOfLength(randomIntBetween(1, 50));
+        ClusterState clusterState = createClusterState(indexName);
+        BlobPath expectedPath = getPath();
+
+        LatchedActionListener<ClusterMetadataManifest.UploadedMetadata> listener = mock(LatchedActionListener.class);
+        when(blobStore.blobContainer(expectedPath)).thenReturn(blobContainer);
+
+        remoteRoutingTableService.start();
+        CheckedRunnable<IOException> runnable = remoteRoutingTableService.getIndexRoutingAsyncAction(
+            clusterState,
+            clusterState.routingTable().getIndicesRouting().get(indexName),
+            listener,
+            basePath
+        );
+        assertNotNull(runnable);
+        runnable.run();
+
+        verify(blobContainer, times(1)).writeBlob(anyString(), any(StreamInput.class), anyLong(), eq(true));
+        verify(listener, times(1)).onResponse(any(ClusterMetadataManifest.UploadedMetadata.class));
```

Comment:
it is possible the thread might not be executed by the time we reach here, can we use assertBusy to avoid flakiness?
```
        assertBusy(() -> {
            # verification
        });

Reply:
`runnable.run()` is a blocking call, so tests shouldn't be flaky. verified by adding sleep in tests as well.